packages/frontend-shared/src/gql-components/HeaderBarContent.vue

```diff
@@ -305,8 +306,9 @@ function shouldShowPrompt (prompt: { slug: string, noProjectId: boolean, interva
   }
 
   const now = Date.now()
-  const timeSinceOpened = now - savedState.value?.firstOpened
+  const timeSinceOpened = now - (savedState.value?.firstOpened ?? now)
   const allPromptShownTimes: number[] = Object.values(savedState.value?.promptsShown ?? {})
+  const bannersLastShown = Object.values(savedState.value?.banners ?? {}).map((banner) => banner?.lastShown).filter((val): val is number => !!val)
```

Comment:
This would be for another issue, but it would be nice to move this "shouldShowPrompt" logic out of the HeaderBarContent component. I know the requirements for this issue call for this additional dependency, but it seems like a strange coupling in the code.

Reply:
Agreed; my thoughts are that these cross-cutting concerns would get bundled up as part of the Banner Framework so that a shared composable could be used. Didn't want to take that on in this work since we hadn't done any planning for what that framework needed to look like